### ISSUE TITLE
关于结果的复现

### ISSUE THREAD

**Issue by willer-lu** [THU-KEG/KB-Plugin#5]
当直接使用你们提供的checkpoint时，推理的结果与论文报告的结果基本上一样。    
但是当我进行完整的训练流程再进行推理时，结果与论文差距很大，我只是将训练schma plugin的相关脚本文件中的bs由128改为了64来适应A100的40G显存。源知识图谱的训练只改了bs，目标知识图谱除了bs也只进行过并行设置（多卡并行）。    
对于webqsp数据集而言，F1只有50.8，对于soaybench而言，f1只有80.7。      
这个结果的波动似乎不太正常，是否是训练阶段引入了什么错误？    
谢谢！！！

**Comment by willer-lu** [comment]
我整个过程都follow了readme，除了没有做metaqa的实验，我认为没有步骤上的错误

**Comment by willer-lu** [comment]
或者你们能提供16个schma plugin的checkpoint吗？非常感谢！！！这个项目我花了挺久的。

**Comment by Neo-Zhangjiajie** [comment]
Hello， we will upload the 16 schema plugins of source KBs soon. 
And I think since you have set the parallel training for the target KBs, the actual batch size should be BS * num_parallel, so the training steps for their schema plugins may be too few. According to our experience, the PI plugin performs the best when the schema plugins of source and target KBs have similar batch sizes and training steps.

**Comment by Neo-Zhangjiajie** [comment]
Also, when you set the batch size for source KB to 64, the total training step would be 36000, which seems too many. You can set the “gradient accumulate step” to 2 to save the GPU memory while keeping the actual batch size 128.

**Comment by Neo-Zhangjiajie** [comment]
The 16 schema plugins of KQA Pro can be found here: https://drive.google.com/drive/folders/1dsCg_zhqaUT9g5NDT0C_L9xTkeTaVUFv?usp=sharing

**Comment by willer-lu** [comment]
> Also, when you set the batch size for source KB to 64, the total training step would be 36000, which seems too many. You can set the “gradient accumulate step” to 2 to save the GPU memory while keeping the actual batch size 128.

谢谢！但是在你们代码中，    

> total_bs=128
> bs=64
> num_gpus=1
> lr=1e-5
> model_name_or_path="../../models_hf/llama-2-7b"
> num_train_epochs=3
> save_steps=1000
> 
> total_per_device=$((${total_bs}/${num_gpus}))
> accu=$(( ${total_per_device} / ${bs} ))

改bs会对应修改gradient accumulation step，总的total_bs是没有改变的，在train_el_lora_0-7.sh等文件中，修改bs从128到64需要修改其他变量或者参数吗？即使修改了nun_gpus比如从1到2（并在localhost指定对应显卡），总的total_bs应该也没有改变，会影响total_step吗？

**Comment by Neo-Zhangjiajie** [comment]
>改bs会对应修改gradient accumulation step，总的total_bs是没有改变的，在train_el_lora_0-7.sh等文件中，修改bs从128到64需要修改其他变量或者参数吗？即使修改了nun_gpus比如从1到2（并在localhost指定对应显卡），总的total_bs应该也没有改变，会影响total_step吗？

You are right. I forgot that we have set the total_bs to be fixed so the total_step will not change. Just in case, you can check whether the final training steps of your trained schema plugins are similar to our released checkpoints.

**Comment by willer-lu** [comment]
> > 改bs会对应修改gradient accumulation step，总的total_bs是没有改变的，在train_el_lora_0-7.sh等文件中，修改bs从128到64需要修改其他变量或者参数吗？即使修改了nun_gpus比如从1到2（并在localhost指定对应显卡），总的total_bs应该也没有改变，会影响total_step吗？
> 
> You are right. I forgot that we have set the total_bs to be fixed so the total_step will not change. Just in case, you can check whether the final training steps of your trained schema plugins are similar to our released checkpoints.

谢谢作者这段时间以来的答复，很抱歉，即使在你们提供的源域的schema的checkpoint上进行pi plugin的训练（这里训练的细节除机器和可能的环境差异外完全一样），与你们提供的目标域的schma的checkpiont进行merge，依然不能复现，结果依然存在差距，我也不清楚哪里出现了问题，暂时先这样吧。
